### PR TITLE
#166841021 User should be able to follow and unfollow each other

### PR DESCRIPTION
**Pull request by Cavdy**
#### What does this PR do?
- User should be able to follow each other

#### Description of Task to be completed?
- To follow go to route `/api/user/follow/userid`
- To unfollow go to route `/api/user/unfollow/userid`

#### How should this be manually tested?
- N/A

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#166841021](https://www.pivotaltracker.com/story/show/166841021)

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A

### CODE REVIEW

**Review by Alpha1202 — APPROVED** [review]
@Cavdy . Nice job here.

**Review by czarjulius — APPROVED** [review]

**Review by codeBlock-1984 — APPROVED** [review]
Great job @Cavdy. Nice work

File: src/controllers/userController.js [hunk 1]
```diff
@@ -106,6 +106,146 @@ class UserController {
...
+          if (findFollow === null) {
+            await Follow.create({
+              userId: token,
+              followId: userId
+            });
+
+            // RETURN SUCCESS IF SUCCESS
+            return res.status(200).json({
+              status: 200,
+              data: 'you just followed this user',
```
Comment (fob413) [inline src/controllers/userController.js]: Which user? Please use a better (more descriptive message here)

Comment (fob413) [reply src/controllers/userController.js]: Update the rest of your response messages in this file to be more descriptive and well-formatted.

File: src/controllers/userController.js [hunk 2]
```diff
@@ -106,6 +106,146 @@ class UserController {
...
+ *@description Follow User
+ * @static
+ * @param {object} req
+ * @param {object} res
+ * @returns {object} res
+ * @memberof UserController
+ */
+  static async followUser(req, res) {
+    try {
+      const { userId } = req.params;
```
Comment (fob413) [inline src/controllers/userController.js]: create a middleware to validate that this user exists, instead of doing it in the controller

**Comment by fob413** [comment]
Your variable names are not descriptive enough. (doesn't mean they should be too long, at most 4 words)

File: src/controllers/userController.js [hunk 3]
```diff
@@ -106,6 +106,146 @@ class UserController {
...
+      if (findUser !== null) {
+        const findFollow = await Follow.findOne({
+          where: {
+            followId: userId
+          }
+        });
+
+        const token = req.user;
+        // CHECK IF USER IS TRYING TO FOLLOW THEMSELVES
+        if (token !== Number(userId)) {
```
Comment (fob413) [inline src/controllers/userController.js]: Create a middleware for this validation

**Pull request merged**